src/Microsoft.DotNet.VersionTools/Dependencies/Submodule/IndicatorPackageSubmoduleUpdater.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using NuGet.Packaging.Core;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.VersionTools.Dependencies.Submodule
+{
+    /// <summary>
+    /// Downloads a package specified from a dependency build info and updates the target git
+    /// submodule to match the version inside.
+    /// </summary>
+    public class IndicatorPackageSubmoduleUpdater : SubmoduleUpdater
+    {
+        private static readonly Lazy<HttpClient> DownloadClient = new Lazy<HttpClient>();
+
+        /// <summary>
+        /// The NuGet v2 base url to use to download the indicator package, without a trailing '/'.
+        /// For example, https://dotnet.myget.org/F/dotnet-core/api/v2/package.
+        /// </summary>
+        public string PackageDownloadBaseUrl { get; set; }
+
+        public string IndicatorPackageId { get; }
+
+        public IndicatorPackageSubmoduleUpdater(string indicatorPackageId)
+        {
+            if (indicatorPackageId == null)
```

Comment:
No, I'll tighten that up with `IsNullOrEmpty` and change to `ArgumentException`.